app/models/email_configuration.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class EmailConfiguration < ApplicationRecord
+
+  validates :email, :password, :user_name, :authentication, :domain, :tls, :address, length: { maximum: 255 }
+
+  validates :email, :password, :user_name, presence: true
+  validates :email, format: { with: Authentication::RE_EMAIL_OK, message: Authentication::MSG_EMAIL_BAD }
+  validates :email, uniqueness: { case_sensitive: false, message: "This email address is already used." }
```

Comment:
I do not know much about email address RFC but are they really case insensitive?

But a quick search tells me:

> Are email addresses case sensitive RFC?
> Yes, according to RFC 2821, the local mailbox (the portion before `@`) is considered case-sensitive. However, e-mail addresses are not usually case-sensitive because of the difficulties and confusion it would cause between users, the server, and the administrator.Jun 7, 2019

Reply:
They are not but we don't validate them according to spec anyway. See
https://github.com/3scale/porta/blob/master/app/lib/authentication.rb#L16

I don't think emails with only case difference is a good idea. If we need at some point, I assume we can easily relax thta restriction. Easier than ensuring uniqueness later at least.